refactor(linter): introduce `LintRunner`

camc314 opened the pull request:
this PR introduces a `LintRunner`​ struct which handles both the oxlint linting, and tsgolint linting components

Sysix commented on crates/oxc_linter/src/lint_runner.rs:
```diff
@@ -0,0 +1,116 @@
+    }
+}
+
+impl LintRunner {
```
Should the language server use this too? It looks like you had this in mind :)
The only difference should be the `DiagnosticSender`, the language server wants the `Message(WithPosition)` with the fix.

camc314 replied:
yeah this is my intention, is the lang server will re-use `LintRuner`, we can possibly move config resolution into this as well and share it between the two

Sysix commented on crates/oxc_linter/src/lint_runner.rs:
```diff
@@ -0,0 +1,116 @@
+        &mut self,
+        files: &[Arc<OsStr>],
+        tx_error: DiagnosticSender,
+        lint_service_options: LintServiceOptions,
```
Any plans why this is a parameter instead of property?  
The language server is atm not that smart to difference between a "soft/hard" changes for a restart of the Linter.

Sysix reviewed (approved):
Because the `LintRunner` is in `oxc_linter`, I dropped some notice about the language server :) LGFM!

camc314 committed:
refactor(linter): introduce `LintRunner` (#14051)

this PR introduces a `LintRunner`​ struct which handles both the oxlint linting, and tsgolint linting components
